### PR TITLE
core/v1: document that topologyKeys requires the ServiceTopology feature gate

### PR DESCRIPTION
**Pull request by andrewsykim**
Signed-off-by: Andrew Sy Kim <kim.andrewsy@gmail.com>

**What type of PR is this?**
/kind documentation
/kind api-change

**What this PR does / why we need it**:
Document that the ServiceTopology feature gate is required to use the topolgoyKeys API. 

Even though we are going to remove this feature in the future, we should document this requirement since there may be users wanting to try to use this feature gate with the gate disabled.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Document that ServiceTopology feature is required to use `service.spec.topologyKeys`.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/apis/core/types.go [hunk 1]
```diff
@@ -3710,6 +3710,7 @@ type ServiceSpec struct {
 	// The special value "*" may be used to mean "any topology". This catch-all
 	// value, if used, only makes sense as the last value in the list.
 	// If this is not specified or empty, no topology constraints will be applied.
+	// This field is alpha-level and is only honored by servers that enable the ServiceTopology feature.
```
Comment (andrewsykim) [inline pkg/apis/core/types.go:3713]: should we just deprecate this now to start the clock?

Comment (thockin) [reply pkg/apis/core/types.go:3713]: Yes, I don't see why not

**Comment by thockin** [comment]
Thanks!

/lgtm
/approve

File: pkg/apis/core/types.go (hunk 1 above)
Comment (fredrickprashanth) [reply pkg/apis/core/types.go:3713]: Why should this be deprecated ? Is ServiceTopology feature not being used anymore ?

Comment (thockin) [reply pkg/apis/core/types.go:3713]: Yes.  It was an alpha design that we decided there are better answers to.  SO we are deprecating the API for it and will eventually remove it.

Comment (fredrickprashanth) [reply pkg/apis/core/types.go:3713]: Can you suggest some alternatives to achieve the same effect ? Basically to preferentially route traffic based on the cluster topology ?

**Comment by thockin** [comment]
Well, you can keep using the alpha for now, but eventually it will just be
automatic.

File: pkg/apis/core/types.go (hunk 1 above)
Comment (fredrickprashanth) [reply pkg/apis/core/types.go:3713]: For posterity, this is being built into kube-proxy as [Topology Aware proxying](https://github.com/kubernetes/enhancements/issues/2030#issuecomment-701056673)

**Pull request merged**